hitl/streaming.py

```diff
@@ -80,27 +80,31 @@ async def test():
 
         # Sample frames over a set time period and verify that no drops are encountered.
         stop = time.time() + STREAM_TEST_DURATION_SECS
-        dropped_batches = 0
-        total_batches = 0
+        dropped_samples = 0
+        total_samples = 0
 
         while time.time() < stop:
-            for (seqnum, _data) in stream.read_frame():
-                num_dropped = sequence_delta(last_sequence, seqnum)
-                total_batches += 1 + num_dropped
+            frame = stream.read_frame()
 
-                if num_dropped:
-                    dropped_batches += num_dropped
-                    logging.warning('Frame drop detected: 0x%08X -> 0x%08X (%d batches)',
-                                    last_sequence, seqnum, num_dropped)
+            num_dropped = sequence_delta(last_sequence, frame.sequence_number)
 
-                last_sequence = seqnum
+            if num_dropped:
+                dropped_samples += num_dropped
+                logging.warning('Frame drop detected: 0x%08X -> 0x%08X (%d batches)',
+                                last_sequence, frame.sequence_number, num_dropped)
 
-        assert total_batches, 'Stream did not receive any frames'
-        stream_efficiency = 1.0 - (dropped_batches / total_batches)
+            num_samples = len(frame.traces[list(frame.traces.keys())[0]])
+
+            total_samples += num_samples + num_dropped
+
+            last_sequence = frame.sequence_number + num_samples - 1
+
+        assert total_samples, 'Stream did not receive any frames'
+        stream_efficiency = 1.0 - (dropped_samples / total_samples)
```

Comment:
This all looks a bit convoluted still. It should use the same logic and preferably code with: https://github.com/quartiq/stabilizer/blob/master/scripts/stream_throughput.py

Reply:
The intent here is that we're interating across each received stream frame instead of each batch - I have forgotten to update `stream_throughput.py`, but that script would no longer work with these changes.

Reply:
I had the impression that we could have merged the code. The `stream_throughput` seemed latest and greatest.